pkg/cassandra/nodetool/nodetool.go

```diff
@@ -0,0 +1,170 @@
+package nodetool
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/pborman/uuid"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/jetstack/navigator/pkg/cassandra/nodetool/client"
+)
+
+type NodeState string
+
+const (
+	NodeStateUnknown NodeState = "Unknown"
+	NodeStateNormal  NodeState = "Normal"
+	NodeStateLeaving NodeState = "Leaving"
+	NodeStateJoining NodeState = "Joining"
+	NodeStateMoving  NodeState = "Moving"
+)
+
+type NodeStatus string
+
+const (
+	NodeStatusUnknown NodeStatus = "Unknown"
+	NodeStatusUp      NodeStatus = "Up"
+	NodeStatusDown    NodeStatus = "Down"
+)
+
+type Node struct {
+	Host   string
+	ID     uuid.UUID
+	State  NodeState
+	Status NodeStatus
+	Local  bool
+}
+
+type NodeMap map[string]*Node
+
+func (nm NodeMap) LocalNode() *Node {
+	for _, node := range nm {
+		if node.Local {
+			return node
+		}
+	}
+	return nil
+}
+
+type Interface interface {
+	Status() (NodeMap, error)
+}
+
+type tool struct {
+	client client.Interface
+}
+
+var _ Interface = &tool{}
+
+func New(client client.Interface) Interface {
+	return &tool{
+		client: client,
+	}
+}
+
+func NewFromURL(u *url.URL) Interface {
+	return New(client.New(u, &http.Client{}))
+}
+
+func setsIntersect(setsToCheck ...sets.String) bool {
+	all := sets.NewString()
+	totalLength := 0
+	for _, s := range setsToCheck {
+		totalLength += s.Len()
+		all = all.Union(s)
+	}
+	return all.Len() != totalLength
+}
+
+func (t *tool) Status() (NodeMap, error) {
+	ssInfo, err := t.client.StorageService()
+	if err != nil {
+		return nil, err
+	}
+
+	nodes := NodeMap{}
+	mappedNodes := sets.NewString()
+	for host, id := range ssInfo.HostIdMap {
+		mappedNodes.Insert(host)
+		nodes[host] = &Node{
+			Host:   host,
+			ID:     id,
+			Status: NodeStatusUnknown,
+			State:  NodeStateNormal,
+		}
+	}
+
+	liveNodes := sets.NewString(ssInfo.LiveNodes...)
+	unreachableNodes := sets.NewString(ssInfo.UnreachableNodes...)
+	if setsIntersect(liveNodes, unreachableNodes) {
+		return nil, fmt.Errorf(
+			"The sets of live and unreachable nodes should not intersect. "+
```

Comment:
drop caps from the start of this error.
you might then want to do the same for Live/Unreachable but idm. convention does not dictate anything beyond the first char